lib/logstash/devutils/rspec/shared_examples.rb

```diff
@@ -0,0 +1,23 @@
+require 'rspec/wait'
+
+RSpec.shared_examples "an interruptible input plugin" do
+  describe "#stop" do
+    let(:queue) { SizedQueue.new(20) }
+    subject do
+      s = described_class.new(config)
```

Comment:
Careful when doing the first. by default subject will both instantiate AND run register, and redefining this will skip register.
I believe I had trouble defining a before block in a shared_example, so that's why it's like this.